### PR TITLE
API Endpoints for collaborators

### PR DESCRIPTION
**Pull request by ethantkoenig**
Adds the following endpoints:
- `GET repos/:username/:reponame/collaborators`: list collaborators
- `GET repos/:username/:reponame/collaborators/:collaborator`: check if a user is a collaborator
- `DELETE repos/:username/:reponame/collaborators/:collaborator`: remove a collaborator

These endpoints are part of [the Github API](https://developer.github.com/v3/repos/collaborators/)

### CODE REVIEW

**Comment by bkcsoft** [comment]
Relevant functions needed in `go-sdk` as well :)

**Comment by lunny** [comment]
But there is no permission check?

**Comment by ethantkoenig** [comment]
@lunny Fixed. I also added a check for the existing `PUT repos/:username/:reponame/collaborators/:collaborator` endpoint.

File: routers/api/v1/repo/collaborators.go [hunk 1]
```diff
@@ -11,8 +11,72 @@ import (
...
 )
 
+// ListCollaborators list a repository's collaborators
+func ListCollaborators(ctx *context.APIContext) {
+	access, err := models.AccessLevel(ctx.User, ctx.Repo.Repository)
+	if err != nil {
+		ctx.Error(500, "AccessLevel", err)
+		return
+	}
+	if access < models.AccessModeWrite {
```
Comment (bkcsoft) [inline routers/api/v1/repo/collaborators.go:21]: This should only be checked for private repos. For public repos anyone can fetch the list of collaborators :)

File: routers/api/v1/repo/collaborators.go [hunk 2]
```diff
@@ -11,8 +11,72 @@ import (
...
+}
+
+// IsCollaborator check if a user is a collaborator of a repository
+func IsCollaborator(ctx *context.APIContext) {
+	access, err := models.AccessLevel(ctx.User, ctx.Repo.Repository)
+	if err != nil {
+		ctx.Error(500, "AccessLevel", err)
+		return
+	}
+	if access < models.AccessModeWrite {
```
Comment (bkcsoft) [inline routers/api/v1/repo/collaborators.go:44]: same as above

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: routers/api/v1/repo/collaborators.go (hunk 1 above)
Comment (ethantkoenig) [reply routers/api/v1/repo/collaborators.go:21]: It is always checked in the Github API, even for public repos.

Comment (bkcsoft) [reply routers/api/v1/repo/collaborators.go:21]: ooh nvm then 😄

**Review by bkcsoft — APPROVED** [review]

**Comment by bkcsoft** [comment]
LGTM

**Comment by lunny** [comment]
LGTM

**Pull request merged**